roles/nextcloud/tasks/install.yml

```diff
@@ -1,183 +1,117 @@
+#     # Ansible's Version Comparison routine:
+#     # https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#version-comparison
+#   when: php_version_installed.stdout != ""    # i.e. IF ABOVE 'php -v' WORKED
+
+# - debug:
+#     var: php_new
+
+
+# https://docs.nextcloud.com/server/18/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation
+- name: Install ffmpeg + libxml2 + 13 php packages
   package:
     name:
-      - libapache2-mod-php
-      - php-imagick
+      #- dnsutils        # NOT REQUESTED by Nextcloud
+      - ffmpeg           # Optional (for preview generation)
+      - libxml2          # php-libxml requires libxml2 >= 2.7.0
+      #- libapache2-mod-php    # 2020-02-15: NO LONGER NEEDED?
+      - php-bz2          # Optional (for extraction of apps)
+      #- php-cli         # NOT REQUESTED by Nextcloud
+      #- php-common      # NOT REQUESTED by Nextcloud
+      - php-curl
+      - php-fpm          # ALSO INSTALLED IN roles/www_back_end (Stage 3) for nginx_high_php_limits
+      - php-gd
+      - php-gmp          # Optional (for SFTP storage)
+      - php-imagick      # Optional (for preview generation)
+      - php-intl         # Optional (increases language translation performance and fixes sorting of non-ASCII characters)
+      - php-json
+      #- php-libxml      # NOT INSTALLABLE: ENABLED BY DEFAULT (https://www.php.net/manual/en/libxml.installation.php)
       - php-mbstring
+      - php-mysql 
+      #- php-openssl     # NOT INSTALLABLE: ENABLED BY DEFAULT?
+      #- php-pdo_mysql   # NOT INSTALLABLE: php-mysql handles this on all OS's?
+      #- php-redis       # @m-anish future work?
+      #- php-session     # NOT INSTALLABLE: ENABLED BY DEFAULT?
+      - php-smbclient    # Optional (SMB/CIFS integration
+      - php-xml          # NOT FORMALLY REQUESTED by Nextcloud (BUT hopefully delivers php-simplexml if not {php-xmlreader, php-xmlwriter} on Raspbian?)
       - php-zip
+      #- php-zlib        # NOT INSTALLABLE: ENABLED BY DEFAULT?
     state: present
-  when: is_ubuntu | bool
 
-- name: Install 5 more php packages (debuntu)
+# https://docs.nextcloud.com/server/18/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation
+- name: Install 9 additional php packages, if OS is not Raspbian (these are not available on Raspbian on RPi, as of Feb 2020)
   package:
     name:
-      - "php{{ php_version }}-gd"
-      - "php{{ php_version }}-json"
-      - "php{{ php_version }}-mysql"
-      - "php{{ php_version }}-curl"
-      - "php{{ php_version }}-intl"
+      - php-ctype
+      - php-dom
+      - php-exif         # Optional (for image rotation in pictures app)
+      - php-fileinfo     # Optional (enhances file analysis performance)
+      - php-iconv
+      - php-posix
+      - php-simplexml
+      - php-xmlreader
+      - php-xmlwriter
     state: present
-  when: is_debuntu | bool
+  when: not is_raspbian
```

Comment:
Given you are seeing 
php-exif is a virtual package provided by:\n php7.3-common 7.3.11-1deb10u1\n
php-fileinfo is a virtual package provided by:\n php7.3-common 7.3.11deb10u1\n
php-ctype is a virtual package provided by:\n php7.3-common 7.3.11-1deb10u1\n
as those modules are now part of common?? Perhaps that could be   
when: not {{ php_version }}| int < 7.3
or php_version is defined in os.yml so you could use the vars that are available there. 
Strange raspbian is not seeing the virtual depends correctly, pity.

Reply:
`php_version` is definitely 7.3 on Raspbian Stretch on RPi, as I've reconfirmed in places like:
https://github.com/iiab/iiab/blob/master/vars/raspbian-10.yml#L29

Does it make sense to add in `php-common` to the list at https://github.com/iiab/iiab/blob/master/roles/nextcloud/tasks/install.yml#L43-L72 to be safe?

Also...is it likely that most all 9 php-* packages at https://github.com/iiab/iiab/blob/master/roles/nextcloud/tasks/install.yml#L74-L88 are gratuitous, having been installed anyway by larger superset php-* packages...such that many or all of these 9 lines can be yanked?

Reply:
> pi@raspberrypi:~ $ sudo apt install php-ctype
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package php-ctype is a virtual package provided by:
  php7.3-common 7.3.11-1~deb10u1
  php7.2-common 7.2.9-1+b2
  php7.1-common 7.1.20-1+b2
You should explicitly select one to install.

> pi@raspberrypi:~ $ sudo apt install php-dom
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package php-dom is a virtual package provided by:
  php7.3-xml 7.3.11-1~deb10u1
  php7.2-xml 7.2.9-1+b2
  php7.1-xml 7.1.20-1+b2
You should explicitly select one to install.

Think you went the wrong way in trying to unify the package names by dropping the use of php_version in the package name of the apt install request.

Reply:
well maybe not, php(7.3)-common does get dragged in.
> pi@raspberrypi:~ $ sudo apt install php-mysql 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  php-common php7.3-common php7.3-mysql
The following NEW packages will be installed:
  php-common php-mysql php7.3-common php7.3-mysql
0 upgraded, 4 newly installed, 0 to remove and 0 not upgraded.
Need to get 635 kB of archives.
After this operation, 6,878 kB of additional disk space will be used.
Do you want to continue? [Y/n] n
Abort.

 I think that list you have is a stale apt listing from an older php version.

Reply:
fyi this helpful discussion is moving / has moved to #2258.